r/R/zzz.R

```diff
@@ -15,6 +15,19 @@
 # specific language governing permissions and limitations
 # under the License.
 
-#' @useDynLib arrow, .registration = TRUE
 #' @importFrom Rcpp sourceCpp
 NULL
+
+NAMESPACE <- environment()
+
+.onLoad <- function(libname, pkgname) {
```

Comment:
@jimhester @jjallaire I'm doing this, inspired from what RcppParallel does, so that I could load the package library after e.g. the dependent Arrow and Parquet libraries, or perhaps gracefully fail at runtime when these are not available

but it seems that this does not play nicely with `load_all()` : 

Is that a known thing ?

```
❯ devtools::load_all()
Loading arrow
Error in find.package(package, lib.loc, verbose = verbose) : 
  there is no package called ‘arrow’
❯ traceback()
7: stop(packageNotFoundError(bad, lib.loc, sys.call()))
6: find.package(package, lib.loc, verbose = verbose)
5: library.dynam("arrow", pkgname, libname) at zzz.R#24
4: nsenv[[f_name]](dirname(ns_path), package)
3: run_pkg_hook(package, "load")
2: pkgload::load_all(path = path, reset = reset, recompile = recompile, 
       export_all = export_all, helpers = helpers, quiet = quiet, 
       ...)
1: devtools::load_all()
```

`devtools::document()` fails too and leaves a dirty NAMESPACE